src/nvim/terminal.c

```diff
@@ -907,6 +928,16 @@ static void on_refresh(Event event)
       }
       continue;
     }
+    FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+      if (wp->w_buffer == term->buf) {
+        if ( wp->w_botline > wp->w_buffer->b_ml.ml_line_count ) {
+          // if last line is visible, follow the terminal
+          wp->following_terminal = true;
+        } else {
+          wp->following_terminal = false;
+        }
```

Comment:
In some situations you don't want the unfocused window to scroll, as you described. However, in some situations you do, e.g. I've started a rather long-running command (e.g. a make) and I want to make some minor changes to other files, while monitoring the output. Then I don't want to have to "jump to the terminal window, go down, jump back" every once in a while to know what's going on, I want the window to scroll.
That's why I chose this solution : if the last line of the output (probably the line where the terminal is adding its stuff) is visible in the vim window, we'll follow what's going on. If you don't want the window to scroll, ensure that the last line is not visible. 
That's behavior I've seen in a number of terminal emulators that I was very happy with :-)